### PR TITLE
Adding resilient code

### PR DESCRIPTION
**Pull request by ajlopez**
It's only a baby step to avoid the permanent stop of top miner, in case of some incidents building the new block to mine (like uncatched exception executing a transaction).

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java [hunk 1]
```diff
@@ -504,7 +506,12 @@ private Block createBlock(Block newBlockParent, List<BlockHeader> uncles, List<T
         @Override
         public void run() {
             Block bestBlock = blockchain.getBestBlock();
-            buildBlockToMine(bestBlock, false);
+            try {
+                buildBlockToMine(bestBlock, false);
+            } catch (Throwable th) {
+                logger.error("Unexpected error: ", th);
+                panicProcessor.panic("mserror", th.toString());
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java]: Please use `th.getMessage()` for consistency.

**Comment by rskops** [comment]
SonarQube analysis reported 16 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 12 major
* ![INFO][INFO] 3 info

#### Top 10 issues

1. ![BLOCKER][BLOCKER] [MinerServerImpl.java#L511](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L511): Catch Exception instead of Throwable. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1181)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L150](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L150): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L161](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L161): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L175](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L175): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L186](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L186): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L189](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L189): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L280](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L280): Boxing/unboxing to parse a primitive co.rsk.mine.MinerServerImpl.getWork() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ADM_BOXED_PRIMITIVE_FOR_PARSING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L280](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L280): Method co.rsk.mine.MinerServerImpl.getWork() converts String to primitive using excessive boxing [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOXING_PARSE)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L300](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L300): Method co.rsk.mine.MinerServerImpl.updateGetWork(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L356](https://github.com/rsksmart/rskj/blob/af62acebfbeb4310317abbcfb0517e09a7dc0ce0/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L356): Method co.rsk.mine.MinerServerImpl.buildBlockToMine(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java (hunk 1 above)
Comment (ajlopezrsk) [reply rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java]: done!

**Comment by rskops** [comment]
SonarQube analysis reported 16 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 12 major
* ![INFO][INFO] 3 info

#### Top 10 issues

1. ![BLOCKER][BLOCKER] [MinerServerImpl.java#L511](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L511): Catch Exception instead of Throwable. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1181)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L150](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L150): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L161](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L161): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L175](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L175): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L186](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L186): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L189](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L189): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L280](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L280): Boxing/unboxing to parse a primitive co.rsk.mine.MinerServerImpl.getWork() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ADM_BOXED_PRIMITIVE_FOR_PARSING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L280](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L280): Method co.rsk.mine.MinerServerImpl.getWork() converts String to primitive using excessive boxing [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOXING_PARSE)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L300](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L300): Method co.rsk.mine.MinerServerImpl.updateGetWork(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L356](https://github.com/rsksmart/rskj/blob/bcce5ff2ab65800cc9ba2daf71892091999a3d5b/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L356): Method co.rsk.mine.MinerServerImpl.buildBlockToMine(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by rskops** [comment]
SonarQube analysis reported 16 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 12 major
* ![INFO][INFO] 3 info

#### Top 10 issues

1. ![BLOCKER][BLOCKER] [MinerServerImpl.java#L511](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L511): Catch Exception instead of Throwable. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1181)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L150](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L150): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L161](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L161): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L175](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L175): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L186](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L186): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L189](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L189): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L280](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L280): Boxing/unboxing to parse a primitive co.rsk.mine.MinerServerImpl.getWork() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ADM_BOXED_PRIMITIVE_FOR_PARSING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L280](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L280): Method co.rsk.mine.MinerServerImpl.getWork() converts String to primitive using excessive boxing [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOXING_PARSE)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L300](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L300): Method co.rsk.mine.MinerServerImpl.updateGetWork(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L356](https://github.com/rsksmart/rskj/blob/fa6781d6292cc85e64c3fe06fc78dfb29a6f14ba/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L356): Method co.rsk.mine.MinerServerImpl.buildBlockToMine(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**